### PR TITLE
Fix some bindings specified by uses keywork not work

### PR DESCRIPTION
**Pull request by rujche**
1. Fix some bindings specified by uses keywork not work.
2. Add message to tell customer when environment variables can be used.

Here is an example screenshot:
> image

I tried to use `console` to output this information. It will change a lot of codes. We'd better discuss with Wei before do this change. Let's keep using `log.Printf` in this PR.

### CODE REVIEW

File: cli/azd/resources/scaffold/templates/resources.bicept [hunk 1]
```diff
@@ -905,6 +904,14 @@ module {{bicepName .Name}} 'br/public:avm/res/app/container-app:0.8.0' = {
...
           {{- end}}
           {{- end}}
+          {{- if .Backend}}
+          {{- range $i, $e := .Backend.Frontends}}
+          {
+            name: '{{upper .Name}}_BASE_URL'
+            value: 'https://{{.Name}}.${containerAppsEnvironment.outputs.defaultDomain}'
+          }
+          {{- end}}
+          {{- end}}
```
Comment (rujche) [inline cli/azd/resources/scaffold/templates/resources.bicept:914]: Backend can use frontend url to set CORS.

File: cli/azd/pkg/project/scaffold_gen.go [hunk 2]
```diff
@@ -224,43 +216,9 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
 		}
 	}
 
-	// create reverse frontends -> backends mapping
```
Comment (rujche) [inline cli/azd/pkg/project/scaffold_gen.go:227]: This comment is not accurate.

File: cli/azd/pkg/project/scaffold_gen.go [hunk 3]
```diff
@@ -224,43 +216,9 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
...
-	for i := range infraSpec.Services {
-		svc := &infraSpec.Services[i]
-		if front, ok := backendMapping[svc.Name]; ok {
-			if svc.Backend == nil {
-				svc.Backend = &scaffold.Backend{}
-			}
-			svc.Backend.Frontends = append(svc.Backend.Frontends, scaffold.ServiceReference{Name: front})
-		}
```
Comment (rujche) [inline cli/azd/pkg/project/scaffold_gen.go:235]: Comment serious: 1.2. This is about `frontends -> backends mapping`

File: cli/azd/pkg/project/scaffold_gen.go [hunk 4]
```diff
@@ -224,43 +216,9 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
...
-		if infraSpec.DbPostgres != nil {
-			svc.DbPostgres = &scaffold.DatabaseReference{
-				DatabaseName: infraSpec.DbPostgres.DatabaseName,
-				AuthType:     infraSpec.DbPostgres.AuthType,
-			}
-		}
```
Comment (rujche) [inline cli/azd/pkg/project/scaffold_gen.go:241]: Comment serious: 1.2. This is not about frontends -> backends mapping

File: cli/azd/pkg/project/scaffold_gen.go [hunk 5]
```diff
@@ -224,43 +216,9 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
...
-				AuthType:     infraSpec.DbMySql.AuthType,
-			}
-		}
-		if infraSpec.DbRedis != nil {
-			svc.DbRedis = &scaffold.DatabaseReference{
-				DatabaseName: "redis",
-			}
-		}
-		if infraSpec.DbCosmosMongo != nil {
-			svc.DbCosmosMongo = &scaffold.DatabaseReference{
```
Comment (rujche) [inline cli/azd/pkg/project/scaffold_gen.go:254]: The original code handles the bindings here, but it does not respect `uses`, and not include all bindings. For example: Its doesn't container service bus and event hubs here.

File: cli/azd/pkg/project/scaffold_gen.go [hunk 6]
```diff
@@ -299,49 +333,15 @@ func mapContainerApp(res *ResourceConfig, svcSpec *scaffold.ServiceSpec, infraSp
...
-		if !ok {
-			return fmt.Errorf("resource %s uses %s, which does not exist", res.Name, use)
-		}
-
-		switch useRes.Type {
-		case ResourceTypeDbMongo:
-			svcSpec.DbCosmosMongo = &scaffold.DatabaseReference{DatabaseName: useRes.Name}
-		case ResourceTypeDbPostgres:
-			svcSpec.DbPostgres = &scaffold.DatabaseReference{DatabaseName: useRes.Name}
-		case ResourceTypeDbRedis:
```
Comment (rujche) [inline cli/azd/pkg/project/scaffold_gen.go:330]: The original code handles the bindings here, too. Here respects `uses`, but not include all bindings. For example: Its doesn't contain mysql / service bus / event hubs here.

File: cli/azd/pkg/project/scaffold_gen.go [hunk 7]
```diff
@@ -270,21 +228,97 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
 	return &infraSpec, nil
 }
 
-func mapContainerApp(res *ResourceConfig, svcSpec *scaffold.ServiceSpec, infraSpec *scaffold.InfraSpec) error {
-	props := res.Props.(ContainerAppProps)
+func mapUses(infraSpec *scaffold.InfraSpec, projectConfig *ProjectConfig) error {
```
Comment (rujche) [inline cli/azd/pkg/project/scaffold_gen.go:231]: In the new function, it handles all `uses` in one place for both backing services (like mysql) and container app (like `web` uses `api`)

File: cli/azd/pkg/project/scaffold_gen.go [hunk 8]
```diff
@@ -270,21 +233,100 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
 	return &infraSpec, nil
 }
 
-func mapContainerApp(res *ResourceConfig, svcSpec *scaffold.ServiceSpec, infraSpec *scaffold.InfraSpec) error {
-	props := res.Props.(ContainerAppProps)
+func mapUses(infraSpec *scaffold.InfraSpec, projectConfig *ProjectConfig,
```
Comment (rujche) [inline cli/azd/pkg/project/scaffold_gen.go:236]: In the new function, it handles all `uses`:
1.  Hosting service uses backing services. Link `api` uses `mysql`.
2. One hosting services uses another hosting service. Like `web` uses `api`.

File: cli/azd/pkg/project/scaffold_gen.go [hunk 9]
```diff
@@ -445,173 +453,182 @@ func getServiceSpecByName(infraSpec *scaffold.InfraSpec, name string) *scaffold.
...
+		userResourceName, usedResourceName))
 	if !isJms {
-		log.Printf("spring.cloud.azure.servicebus.namespace")
+		(*console).Message(*context, "spring.cloud.azure.servicebus.namespace")
 	}
 	if authType == internal.AuthTypeUserAssignedManagedIdentity {
-		log.Printf("spring.cloud.azure.servicebus.connection-string=''")
-		log.Printf("spring.cloud.azure.servicebus.credential.managed-identity-enabled=true")
-		log.Printf("spring.cloud.azure.servicebus.credential.client-id")
+		(*console).Message(*context, "spring.cloud.azure.servicebus.connection-string=''")
```
Comment (saragluna) [inline cli/azd/pkg/project/scaffold_gen.go:565]: I am not comfortable with these specific properties are listing directly here, seems like scaffold_gen is a general place to generate the scaffold, but all these properties are only java and spring specific

Comment (rujche) [reply cli/azd/pkg/project/scaffold_gen.go:565]: > but all these properties are only java and spring specific

No, properties like `POSTGRES_USERNAME` are used for all kinds of application, like .NET. The property list used in `scafold_gen.go` is make according to `resources.bicept`.

We can improve this logic in `scafole_gen.go` together with `resourcec.bicept` in this task:  [ADO#SJAD#2300813: User can use spring boot related features (like service connector and environment variables) only when the app is detected as spring boot app](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2300813)

File: cli/azd/pkg/project/scaffold_gen.go [hunk 10]
```diff
@@ -224,67 +221,157 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
...
+				if err != nil {
+					return err
+				}
+			case ResourceTypeDbRedis:
+				printHintsAboutUseRedis(console, context)
+			case ResourceTypeDbMongo:
+				printHintsAboutUseMongo(console, context)
+			case ResourceTypeDbCosmos:
+				printHintsAboutUseCosmos(console, context)
+			case ResourceTypeMessagingServiceBus:
```
Comment (saragluna) [inline cli/azd/pkg/project/scaffold_gen.go:326]: Should we consider language usage here?

**Pull request merged**